base/src/main/scala/typeclass/InvariantFunctor.scala

```diff
@@ -0,0 +1,10 @@
+package scalaz
+package typeclass
+
+trait InvariantFunctor[F[_]] {
+  def xmap[A, B](ma: F[A])(f: A => B)(g: B => A): F[B]
```

Comment:
Since it's called `InvariantFunctor` and not `XFunctor` or `ExpFunctor`, I might suggest `imap` (or possibly just `map` depending on the policy for overloading).

The `(f: A => B)(g: B => A)` may be more usefully represented as `Iso[A, B]` / `<=>`, which composes better and may already be lying around for other purposes.